### PR TITLE
Implement Network Configurators (nmc, nmstate, nmconnections)

### PR DESCRIPTION
**Pull request by anmazzotti**

### CODE REVIEW

File: pkg/network/network.go [hunk 1]
```diff
@@ -31,7 +31,9 @@ import (
 )
 
 const (
-	nmstateTempPath      = "/tmp/elemental-nmstate.yaml"
+	nmcDesiredStatesDir  = "/tmp/desired-states"
+	nmcNewtorkConfigDir  = "/tmp/network-config"
```
Comment (fgiudici) [inline pkg/network/network.go:35]: nit: would consider grouping nmc dirs under a common /tmp/something dir (e.g., "/tmp/nmc/desired-states", "/tmp/nmc/network-config")

**Review by fgiudici — COMMENTED** [review]
Patch looks good... anyway, why not support both nmc and nmstate?
The library behind it is the same, ti changes "just" the binary and the syntax, changes to the code could be minimal.
This would open to wider usage in community images where nmstate could be already be available or packaged for some base OS images, while supporting also the nmc binary where nmstate is not available (or nmc is preferred).
The idea could be to have a kind of "autodetection" just probing for the nmc binary first.
How would that look?

**Comment by anmazzotti** [comment]
> Patch looks good... anyway, why not support both nmc and nmstate? The library behind it is the same, ti changes "just" the binary and the syntax, changes to the code could be minimal. This would open to wider usage in community images where nmstate could be already be available or packaged for some base OS images, while supporting also the nmc binary where nmstate is not available (or nmc is preferred). The idea could be to have a kind of "autodetection" just probing for the nmc binary first. How would that look?

Could be an idea, but I would avoid any autodetection for simplicity. 
We could have a `configurator.type` field that defaults to `nmc` but allows `nmstate` and at this point I can reintroduce the `nmconnections` one, since the implementation is also very similar.

**Comment by anmazzotti** [comment]
@fgiudici I added `network.configurator` that can accept `nmc` (default), `nmstate`, `nmconnections` values.  

I'd also like to install both nmc and nmstate in the dev image, so that we can locally test with both and for example have three-flavors of sample registrations.

Documentation should also be updated accordingly (so most likely one subpage for each configurator, with a few examples)

I'll work on that if you are ok with this solution

**Review by fgiudici — APPROVED** [review]
Having the three options looks quite nice!
Thanks for going the extra mile and adding the network configuration option 💚 
LGTM!

**Pull request merged**